src/app/candidate-management/candidate-management-detail/candidate-management-detail.component.ts

```diff
@@ -107,19 +104,21 @@ export class CandidateManagementDetailComponent implements OnInit, AfterViewInit
   }
 
   ngOnDestroy(): void {
-    this.arSubscription.unsubscribe();
+    if (this.activeRouteSubscription != null) {
+      this.activeRouteSubscription.unsubscribe();
+    }
   }
 
   ngAfterViewInit(): void {
     this.setCommentSectionHeight();
   }
 
-  setupPLChangeDialog() {
-    this.confirmDialog = {
+  public get confirmDialog() {
```

Comment:
```suggestion
  public get confirmDialogData() {
```
Rename to better reflect what the method does (alt.: getConfirmDialogTites/Text)